File: MyPeopleHub.UI/services/AccountService.py
from flask import jsonify;
import requests;

api_url = 'https://localhost:7030'

def login(loginDto):
    try:
        response = requests.post(api_url + '/login', json=loginDto)

        if response.status_code == 200:
            data = response.json()
            return jsonify(data)
        else:
            return jsonify({'error': 'Błąd żądania HTTP'}), response.status_code
    except Exception as e:
        return jsonify({'error': str(e)}), 500
    
def register(registerDto):
    try:
        response = requests.post(api_url + '/register', json=registerDto)

        if response.status_code == 200:
            data = response.json()
            return jsonify(data)
        else:
            return jsonify({'error': 'Błąd żądania HTTP'}), response.status_code
    except Exception as e:
        return jsonify({'error': str(e)}), 500

File: MyPeopleHub.UI/src/app/services/account.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { RegisterDto } from '../models/RegisterDto';
import { BehaviorSubject, Observable, ReplaySubject, Subject } from 'rxjs';
import { LoginDto } from '../models/LoginDto';
import { LoginResponse } from '../models/LoginResponse';
import { TokenService } from './token.service';
import { AccountDataService } from './data-services/account-data.service';

@Injectable({
  providedIn: 'root'
})
export class AccountService {

  private apiUrl = 'http://localhost:5131';

  constructor(private httpClient: HttpClient, private tokenService: TokenService, private accountDataService: AccountDataService) {

  }

  register(registerDto: RegisterDto): Observable<string> {
    return this.httpClient.post<string>(this.apiUrl + '/register', registerDto);
  }

  login(loginDto: LoginDto): Observable<LoginResponse> {
    return this.httpClient.post<LoginResponse>(this.apiUrl + '/login', loginDto);
  }

  logout(): void {
    this.tokenService.removeToken();
    this.accountDataService.setFullName(null);
    this.accountDataService.setLoginStatus(false);
    this.accountDataService.setUserId(null);
  }
}
